pkgs/development/compilers/hare/hare.nix

```diff
@@ -0,0 +1,46 @@
+{ binutils-unwrapped, fetchFromSourcehut, git, harec, lib, makeWrapper, openssh
+, qbe, scdoc, stdenv, }:
+
+stdenv.mkDerivation rec {
+  pname = "hare";
+  version = "2022042601";
+
+  src = fetchFromSourcehut {
+    owner = "~sircmpwn";
+    repo = pname;
+    rev = "a9c144a8d09cbdfee82857f926a413fea0e57298";
+    sha256 = "1z1ryqkg4vswzbnabw2nk7bdl1jjlz3jfqw8h3nylc5mr2zh3chx";
+  };
+
+  configurePhase = ''
+    cp config.example.mk config.mk
+    sed -e "s|^PREFIX = .*|PREFIX = $out|" \
+        -e "s|^LOCALSRCDIR = .*|LOCALSRCDIR = $out/local/src/hare|" \
+        -i config.mk
+  '';
+
+  # Default cache location is user home, not writeable at Nix build time.
+  preBuild = ''
+    export HARECACHE="$NIX_BUILD_TOP"
+  '';
+  buildInputs = [ qbe harec scdoc makeWrapper ];
```

Comment:
At least `scdoc` and `makeWrapper` look to me like they should be native build inputs.  Try cross-compiling or setting `strictDeps = true;` (good practice anyway) to check.